dotnet/src/SemanticKernel.Abstractions/SemanticFunctions/PromptTemplateConfig.cs

```diff
@@ -63,43 +64,74 @@ public class InputConfig
     public int Schema { get; set; } = 1;
 
     /// <summary>
-    /// Type, such as "completion", "embeddings", etc.
+    /// Description
     /// </summary>
-    /// <remarks>TODO: use enum</remarks>
-    [JsonPropertyName("type")]
+    [JsonPropertyName("description")]
     [JsonPropertyOrder(2)]
-    public string Type { get; set; } = "completion";
+    public string Description { get; set; } = string.Empty;
 
     /// <summary>
-    /// Description
+    /// Input configuration (that is, list of all input parameters).
     /// </summary>
-    [JsonPropertyName("description")]
+    [JsonPropertyName("input")]
     [JsonPropertyOrder(3)]
-    public string Description { get; set; } = string.Empty;
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public InputConfig Input { get; set; } = new();
 
     /// <summary>
-    /// Completion configuration parameters.
+    /// Model request settings.
     /// </summary>
-    [JsonPropertyName("completion")]
+    [JsonPropertyName("models")]
     [JsonPropertyOrder(4)]
     [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
-    public AIRequestSettings? Completion { get; set; }
+    public List<AIRequestSettings> Models { get; set; } = new();
 
     /// <summary>
-    /// Default AI services to use.
+    /// Return the default AIRequestSettings
     /// </summary>
-    [JsonPropertyName("default_services")]
+    public AIRequestSettings DefaultRequestSettings()
+    {
+        return this.Models.FirstOrDefault<AIRequestSettings>();
```

Comment:
Shouldn't we use `.Models.SingleOrDefault(model => string.IsNullOrEmpty(model.ServiceId))` to identify default request settings rather than grabbing the first one?

Reply:
This will break when you have more than one model (the proposal is to be multiple model capable)

Reply:
Yes, right, that's the point - there should not be more than one model registered with the **same id**, right?

Reply:
For now (and v1.0) we are allowing multiple models to be registered but we only will use the first one. I have updated the API documentation to reflect this. In a future release we will add the ability to specify multiple models and have the appropriate one selected. So consider this change as future proofing our API to allow for this.